dataforest/core/ProcessRun.py

```
@@ -34,20 +37,24 @@ def forest(self) -> "DataForest":
         current node should only be used for paths, as the parent DataForest is
         returned.
         """
-        # TODO: silent failure prone
-        if not self._forest:
-            self._forest = self._parent_forest.at(self.process_name)
         return self._forest
 
-    @property
-    def forest_any(self):
-        """quick access if it doesn't matter whether forest or parent forest"""
-        return self._forest if self._forest else self._parent_forest
-
     @property
     def path(self) -> Path:
         """Path to directory containing processes output files and logs"""
-        return self.forest_any.paths[self.process_name]
+        return self.forest.paths[self.process_name]
+
+    @property
+    def process_meta(self):
+        """
+        Gets additional metadata created by the current process, which is
+        usually used to merge with the overall metadata.
+        """
+        meta_path = self.path / "meta.tsv"
+        df = None
+        if meta_path.exists():
+            df = pd.read_csv(meta_path, sep="\t", index_col=0)
+        return df
```

If meta for some reason does not exist, this will silently pass on `None`. Maybe:
```
if meta_path.exists():
    df = pd.read_csv(meta_path, sep="\t", index_col=0)
else:
    raise FileNotFoundError(f'No such file or directory: "{meta_path}"')
```

Then, we would also not need `df = None` before, because variables created inside `if` will be visible to whole function.

---

Passing `None` was the intent in the case that there's no metadata, then I wrap with `if __ is not None`, but you're right, that's probably more error prone. I should be doing what you're suggesting and wrap in `try/except`s instead.